diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5cbd77aed6..974efebebc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,3 +58,5 @@
 # NOTE: Policy changes has to happen before adding any subprojects
 cmake_policy(SET CMP0091 NEW)
-set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+if(NOT DEFINED CMAKE_MSVC_RUNTIME_LIBRARY)
+  set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+endif()
diff --git a/third_party/UHDM/CMakeLists.txt b/third_party/UHDM/CMakeLists.txt
index 3425264994..3bc86df0a1 100644
--- a/third_party/UHDM/CMakeLists.txt
+++ b/third_party/UHDM/CMakeLists.txt
@@ -31,3 +31,5 @@
 # NOTE: Policy changes has to happen before adding any subprojects
 cmake_policy(SET CMP0091 NEW)
-set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+if(NOT DEFINED CMAKE_MSVC_RUNTIME_LIBRARY)
+  set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+endif()
